### PR TITLE
test: Move TestMultiOSDirect from centos-8-stream to rhel-8-10, drop CentOS 8 special cases

### PR DESCRIPTION
**Pull request by martinpitt**
I accidentally [broke main](https://cockpit-logs.us-east-1.linodeobjects.com/pull-20425-d3594a54-20240507-104057-debian-stable-other/log.html#49-2) with https://github.com/cockpit-project/bots/pull/6348 , sorry about that!

### CODE REVIEW

File: test/verify/check-networkmanager-wireguard [hunk 1]
```diff
@@ -30,7 +30,7 @@ class TestWireGuard(packagelib.PackageCase, netlib.NetworkCase):
         b.wait_visible("#network-wireguard-settings-dialog")
         iface_name = b.val("#network-wireguard-settings-interface-name-input")
         b.wait_visible("#network-wireguard-settings-save:disabled")
-        if m1.image in ["rhel4edge", "centos-8-stream"] or m1.image.startswith("rhel-8"):
+        if m1.image in ["rhel4edge"] or m1.image.startswith("rhel-8"):
```
Comment (allisonkarlitskaya) [inline test/verify/check-networkmanager-wireguard:33]: I've seen `ruff` complain about stuff like this before...

File: test/verify/check-shell-multi-os [hunk 2]
```diff
@@ -85,11 +85,11 @@ class TestRHEL8(testlib.MachineCase):
 class TestMultiOSDirect(testlib.MachineCase):
     provision = {
         "0": {"address": "10.111.113.1/20", "memory_mb": 512},
-        "stock": {"address": "10.111.113.5/20", "image": "centos-8-stream", "memory_mb": 512}
```
Comment (allisonkarlitskaya) [inline test/verify/check-shell-multi-os:88]: bzzt bzzt  oops :)

**Review by allisonkarlitskaya — APPROVED** [review]
:+1: thanks!

File: test/verify/check-networkmanager-wireguard (hunk 1 above)
Comment (martinpitt) [reply test/verify/check-networkmanager-wireguard:33]: OOI, what about?

Comment (martinpitt) [reply test/verify/check-networkmanager-wireguard:33]: Ah! Can I say "let's use this as a reminder to kill it off completely"? :grin: At some point rhel4edge ought to catch up with RHEL 9.4 (it was just released in fact, so we should soon get a 9.4 based image).

**Pull request merged**